Library/Homebrew/check_ruby_version.rb

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby --enable-frozen-string-literal --disable=gems,did_you_mean,rubyopt
+# frozen_string_literal: true
+
+HOMEBREW_REQUIRED_RUBY_VERSION = ARGV.first.freeze
+raise "No Ruby version passed!" if HOMEBREW_REQUIRED_RUBY_VERSION.to_s.empty?
+
+require "rubygems"
+
+ruby_version = Gem::Version.new(RUBY_VERSION)
+homebrew_required_ruby_version = Gem::Version.new(HOMEBREW_REQUIRED_RUBY_VERSION)
+
+ruby_version_major, ruby_version_minor, = ruby_version.canonical_segments
+homebrew_required_ruby_version_major, homebrew_required_ruby_version_minor, =
+  homebrew_required_ruby_version.canonical_segments
```

Comment:
I see this when I use it on a system with old system Ruby: 
```
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/check_ruby_version.rb:12:in `<main>': undefined method `canonical_segments' for #<Gem::Version "2.3.1"> (NoMethodError)
```

Reply:
We can mimick this with:
```suggestion
ruby_version_major, ruby_version_minor, = ruby_version.to_s.split('.').map(&:to_i)
homebrew_required_ruby_version_major, homebrew_required_ruby_version_minor, =
  homebrew_required_ruby_version.to_s.split('.').map(&:to_i)
```

Reply:
Does `.segments` work? Otherwise I'm tempted to have `abort unless ruby_version.respond_to?(:canonical_segments)` because it's too old to work anyway.

Reply:
Did that!